### PR TITLE
Expose codegen

### PR DESCRIPTION
**Pull request by oovm**

### CODE REVIEW

**Comment by oovm** [comment]
Can a new version be released to get recent update?

File: lrpeg-example/Cargo.toml [hunk 1]
```diff
@@ -2,11 +2,11 @@
...
+edition = "2021"
 
 [build-dependencies.lrpeg]
 path = "../lrpeg"
 
 [dependencies]
-regex = "1"
-unicode-xid = "0.2"
+regex = "1.5.4"
+unicode-xid = "0.2.2"
```
Comment (seanyoung) [inline lrpeg-example/Cargo.toml:12]: Why pin the dependencies to specific versions?

When new versions of the crates are released, they should just work (if they follow semver correctly).

File: lrpeg-test/Cargo.toml [hunk 2]
```diff
@@ -2,12 +2,12 @@
...
 
 [build-dependencies]
 lrpeg = { path = "../lrpeg" }
 
 [dependencies]
 lrpeg = { path = "../lrpeg" }
-regex = "1"
-unicode-xid = "0.2"
+regex = "1.5.4"
+unicode-xid = "0.2.2"
```
Comment (seanyoung) [inline lrpeg-test/Cargo.toml:13]: same as above

File: lrpeg/Cargo.toml [hunk 3]
```diff
@@ -6,11 +6,11 @@ description = "Left recursive PEG parser generator"
...
+edition = "2021"
 readme = "../README.md"
 license = "MIT"
 workspace = ".."
 
 [dependencies]
-unicode-xid = "0.2"
-regex = "1"
+unicode-xid = "0.2.2"
+regex = "1.5.4"
```
Comment (seanyoung) [inline lrpeg/Cargo.toml:16]: same as above

File: lrpeg/src/peg.peg [hunk 4]
```diff
@@ -35,4 +35,4 @@ regex <- re#re\#([^\#\\]|\\.)*\##;
 
 ws <- (COMMENT / WHITESPACE)*;
 
-COMMENT <- re#//[^\r\n]*# .
+COMMENT <- re#//[^\r\n]*#
```
Comment (seanyoung) [inline lrpeg/src/peg.peg:38]: This dot eats up the linefeed `\n` or `\r`. Not sure why removing it is a good thing. If anything the dot could be moved into the regex.

**Review by seanyoung — COMMENTED** [review]
Just out of interesting, why do you want to expose the codegen? What are you trying to do?

**Comment by oovm** [comment]
I've made some codegen here: [yggdrasil-rs/src/codegen/target_peg](https://github.com/ygg-lang/yggdrasil-rs/blob/master/projects/ygg-core/src/codegen/target_peg/mod.rs)

**Pull request merged**